File: tests/machine/x/pascal/string_contains.pas
program StringContains;
{$mode objfpc}
{$modeswitch nestedprocvars}

uses SysUtils, fgl, fphttpclient, Classes, Variants, fpjson, jsonparser, fpjsonrtti;

type
  generic TArray<T> = array of T;

var
  s: string;

begin
  s := 'catch';
  writeln(s.contains('cat'));
  writeln(s.contains('dog'));
end.

File: tests/transpiler/x/pl/string_contains.pl
:- style_check(-singleton).
:- initialization(main).

main :-
    S = 'catch',
    (sub_string(S, _, _, _, 'cat') -> writeln(true) ; writeln(false)),
    (sub_string(S, _, _, _, 'dog') -> writeln(true) ; writeln(false)).
